src/button-dropdown/category-elements/category-element.tsx

```diff
@@ -12,10 +12,12 @@ const CategoryElement = ({
   onGroupToggle,
   targetItem,
   isHighlighted,
+  isFocusOn,
```

Comment:
What is the meaning of this flag? It is not clear from the code what is supposed to be in focus. Also, when real focusing is used it is normally possible to determine if the element or some of its sub-elements is focused, e.g.:

```
if (elementRef.current === document.activeElement || elementRef.current?.contains(document.activeElement)) {
  // focused
}
```